loader.js

```diff
@@ -34,12 +35,20 @@ module.exports = async function mdxEnhancedLoader(src) {
 
   // Checks if there's a layout, if there is, resolve the layout and wrap the content in it.
   processLayout
-    .call(this, options, data, content, resourcePath)
+    .call(this, options, data, content, resourcePath, scans)
     .then(result => callback(null, result))
     .catch(err => callback(err))
 }
-
-function processLayout(options, frontMatter, content, resourcePath) {
+function scanContent(options, content) {
+  const { mdxEnhancedPluginOptions: pluginOpts } = options
+  if (typeof pluginOpts.scan === 'undefined') return {}
+  return Object.keys(pluginOpts.scan).reduce((acc, opt) => {
+    acc[opt] = content.match(pluginOpts.scan[opt].pattern)
```

Comment:
I guess if you wanted the ability for user to provide some sort of extraction mechanism by way of function call, like a `value` key you might be able to implement it like this:

```js
acc[opt] = typeof pluginOpts.scan[opt].value === 'function'
    ? pluginOpts.scan[opt].value(content.match(pluginOpts.scan[opt].pattern))
    : content.match(pluginOpts.scan[opt].pattern)
```

Reply:
Thanks @pruett ! I had something like this prior. So absolutely it makes sense if we want to support multiple `value` param types. 

The more I thought about it though, we probably won't need `value` if not provided as a function anyway, given that we'll need to check for presence of a match upfront anyway.. Here's where I landed: 
```js
  return Object.keys(pluginOpts.scan).reduce((acc, scanName) => {

    const opt = pluginOpts.scan[scanName]

    // Put the result of the pattern match onto the `scans` object: `{ key : result }`
    if (content.match(opt.pattern)) {
      acc[scanName] = opt.value
        ? opt.value(content.match(opt.pattern)) // We expect a `value` function provided in `next.config.js` if picking the return value is preferred
        : content.match(opt.pattern) // Otherwise pass the Array of matches as the result per key. More info here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match#Return_value
    }
    return acc
  }, {})
}
```

Reply:
@JMFURY I *think* you'll still want the function check to avoid an error in the case that the user provides something that's not callable.

Additionally I think a better name for `value` would be `transform`, which should be documented along the lines of "An optional callback function that transforms the result of the `match` operation" or something...

Of course, you could just forgo the hassle of this sort of transformation altogether and simply return the `match` result as is, putting the responsibility on the developer to transform the result in their application logic...

I'm not familiar enough with the ultimate end use case to get a good sense on what is the right call there. I can see value in providing the transformation option especially if there will be multiple keys and a variety of needs per key etc.

Reply:
Thanks for all this @pruett - I like `transform`, it seems more explicit